### ISSUE TITLE
mfakto errors (sorry to write here, there is no issues button on preda/mfakto)

### ISSUE THREAD

**Issue by valeriob01** [preda/gpuowl#153]
this is the run:
```
./mfakto -d 02 --perftest 1
mfakto 0.15pre6 (64bit build)


Runtime options
  Inifile                   mfakto.ini
  Verbosity                 1
  SieveOnGPU                yes
  MoreClasses               yes
  GPUSievePrimes            81157
  GPUSieveProcessSize       24Ki bits
  GPUSieveSize              96Mi bits
  FlushInterval             0
  WorkFile                  worktodo.txt
  ResultsFile               results.txt
  Checkpoints               enabled
  CheckpointDelay           300s
  Stages                    enabled
  StopAfterFactor           class
  PrintMode                 compact
  V5UserID                  none
  ComputerID                none
  TimeStampInResults        yes
  VectorSize                2
  GPUType                   AUTO
  SmallExp                  no
  UseBinfile                mfakto_Kernels.elf
Select device - Get device info:
WARNING: Unknown GPU name, assuming GCN. Please post the device name "gfx900 (Advanced Micro Devices, Inc.)" to http://www.mersenneforum.org/showthread.php?t=15646 to have it added to mfakto. Set GPUType in mfakto.ini to select a GPU type yourself to avoid this warning.

OpenCL device info
  name                      gfx900 (Advanced Micro Devices, Inc.)
  device (driver) version   OpenCL 2.0  (3098.0 (HSA1.1,LC))
  maximum threads per block 1024
  maximum threads per grid  1073741824
  number of multiprocessors 64 (4096 compute elements)
  clock rate                1630MHz

Automatic parameters
  threads per grid          2097152
  optimizing kernels for    GCN

Compiling kernels.


Perftest

Generate list of the first 1075766 primes: 108.33 ms

1. CPU-Sieve-Init (once per class, 960 times per test, avg. for 1 iterations)
	Init_class(sieveprimes=   5000):     0.40 ms
	Init_class(sieveprimes=  20000):     1.64 ms
	Init_class(sieveprimes=  80000):     7.17 ms
	Init_class(sieveprimes= 200000):    18.97 ms
	Init_class(sieveprimes= 500000):    50.16 ms
	Init_class(sieveprimes=1000000):   104.46 ms

2. CPU-Sieve (output rate M/s)
Sieve size is fixed at compile time, cannot test with variable sizes. Just running 3 fixed tests.

SievePrimes:     254     396     611     945    1460    2257    3487    5389    8328   12871   19890   30738   47503   73411  113449  175323  270944  418716  647083 1000000
SieveSizeLimit
    36 kiB     565.9   512.9   470.3   437.0   397.7   364.6   334.4   308.9   281.8   253.9   223.1   198.5   163.1   129.9   104.4    84.9    69.2    55.2    43.6    34.7
    36 kiB     568.4   522.5   480.4   437.9   398.2   366.2   339.6   310.3   281.8   252.1   221.4   200.0   162.8   127.0   103.0    84.8    68.8    55.2    44.8    35.6
    36 kiB     568.7   519.2   474.0   434.2   398.9   365.3   339.5   313.8   285.0   253.3   225.5   203.2   167.1   129.4   104.7    84.0    66.1    55.1    44.2    35.1

Best SieveSizeLimit for
SievePrimes:     254     396     611     945    1460    2257    3487    5389    8328   12871   19890   30738   47503   73411  113449  175323  270944  418716  647083 1000000
at kiB:           36      36      36      36      36      36      36      36      36      36      36      36      36      36      36      36      36      36      36      36
max M/s:       568.7   522.5   480.4   437.9   398.9   366.2   339.6   313.8   285.0   253.9   225.5   203.2   167.1   129.9   104.7    84.9    69.2    55.2    44.8    35.6
Survivors:    36.41%  34.06%  32.06%  30.28%  28.69%  27.27%  26.00%  24.84%  23.78%  22.82%  21.94%  21.12%  20.36%  19.66%  19.01%  18.40%  17.83%  17.29%  16.79%  16.32%
removal rate   993.2  1011.7  1018.1  1008.5   991.4   976.5   966.5   949.6   913.2   859.1   802.7   759.2   653.5   531.1   445.8   376.6   318.9   264.0   221.9   182.3


3. Memory copy to GPU (blocks of 8388608 bytes)

  Standard copy, standard queue:
      80 MB in   16.9 ms (4970.1 MB/s) (real)

  Standard copy, profiled queue:
      80 MB in   17.2 ms (4886.5 MB/s) (real)
      80 MB in   17.1 ms (4891.8 MB/s) (profiled data)
       8 MB in    1.6 ms (5232.9 MB/s) (profiled data, peak)

  Standard copy, two queues:
      80 MB in   13.3 ms (6315.3 MB/s) (real)

Reinitializing with gpu_sieving enabled.
Select device - Get device info:

OpenCL device info
  name                      gfx900 (Advanced Micro Devices, Inc.)
  device (driver) version   OpenCL 2.0  (3098.0 (HSA1.1,LC))
  maximum threads per block 1024
  maximum threads per grid  1073741824
  number of multiprocessors 64 (4096 compute elements)
  clock rate                1630MHz

Automatic parameters
  threads per grid          2097152
  optimizing kernels for    GCN

Compiling kernels.

4. GPU sieve, 1 iterations each
  GPUSievePrimes (adjusted) 52534
  GPUsieve minimum exponent 646182

 gpusieve_init: 31.842000 ms (CPU work)
 gpusieve_init_exponent: 0.080500 ms (CalcModularInverses)
Memory access fault by GPU node-2 (Agent handle: 0x565557705910) on address 0x7efcf6674000. Reason: Page not present or supervisor privilege.
Aborted
```

this is the compilation:

```
make
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -funroll-all-loops -funsafe-loop-optimizations -fira-region=all -fsched-spec-load -fsched-stalled-insns=10 -fsched-stalled-insns-dep=10 -fno-align-labels -c sieve.c -o sieve.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -c timer.c -o timer.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -c parse.c -o parse.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -c read_config.c -o read_config.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -c mfaktc.c -o mfaktc.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -c checkpoint.c -o checkpoint.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -c signal_handler.c -o signal_handler.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -c filelocking.c -o filelocking.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL -c output.c -o output.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL  -c mfakto.cpp -o mfakto.o
mfakto.cpp: In function ‘int init_CL(int, cl_int*)’:
mfakto.cpp:553:83: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
   commandQueue = clCreateCommandQueue(context, devices[*devnumber], props, &status);
                                                                                   ^
In file included from my_types.h:25,
                 from mfakto.h:23,
                 from mfakto.cpp:25:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
mfakto.cpp:553:83: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
   commandQueue = clCreateCommandQueue(context, devices[*devnumber], props, &status);
                                                                                   ^
In file included from my_types.h:25,
                 from mfakto.h:23,
                 from mfakto.cpp:25:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
mfakto.cpp:557:85: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
     commandQueue = clCreateCommandQueue(context, devices[*devnumber], props, &status);
                                                                                     ^
In file included from my_types.h:25,
                 from mfakto.h:23,
                 from mfakto.cpp:25:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
mfakto.cpp:557:85: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
     commandQueue = clCreateCommandQueue(context, devices[*devnumber], props, &status);
                                                                                     ^
In file included from my_types.h:25,
                 from mfakto.h:23,
                 from mfakto.cpp:25:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
mfakto.cpp:571:86: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
   commandQueuePrf = clCreateCommandQueue(context, devices[*devnumber], props, &status);
                                                                                      ^
In file included from my_types.h:25,
                 from mfakto.h:23,
                 from mfakto.cpp:25:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
mfakto.cpp:571:86: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
   commandQueuePrf = clCreateCommandQueue(context, devices[*devnumber], props, &status);
                                                                                      ^
In file included from my_types.h:25,
                 from mfakto.h:23,
                 from mfakto.cpp:25:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
mfakto.cpp: In function ‘int run_mod_kernel(cl_ulong, cl_ulong, cl_ulong, cl_float, cl_ulong*, cl_ulong*)’:
mfakto.cpp:1654:26: warning: ‘cl_int clEnqueueTask(cl_command_queue, cl_kernel, cl_uint, _cl_event* const*, _cl_event**)’ is deprecated [-Wdeprecated-declarations]
                  &mod_evt);
                          ^
In file included from my_types.h:25,
                 from mfakto.h:23,
                 from mfakto.cpp:25:
/opt/rocm/opencl/include/CL/cl.h:1378:1: note: declared here
 clEnqueueTask(cl_command_queue  /* command_queue */,
 ^~~~~~~~~~~~~
mfakto.cpp:1654:26: warning: ‘cl_int clEnqueueTask(cl_command_queue, cl_kernel, cl_uint, _cl_event* const*, _cl_event**)’ is deprecated [-Wdeprecated-declarations]
                  &mod_evt);
                          ^
In file included from my_types.h:25,
                 from mfakto.h:23,
                 from mfakto.cpp:25:
/opt/rocm/opencl/include/CL/cl.h:1378:1: note: declared here
 clEnqueueTask(cl_command_queue  /* command_queue */,
 ^~~~~~~~~~~~~
mfakto.cpp: In function ‘int run_kernel15(cl_kernel, cl_uint, int75, int, cl_uint8, cl_mem, cl_int, cl_int)’:
mfakto.cpp:1722:5: note: The ABI for passing parameters with 32-byte alignment has changed in GCC 4.6
 int run_kernel15(cl_kernel l_kernel, cl_uint exp, int75 k_base, int stream, cl_uint8 b_in, cl_mem res, cl_int shiftcount, cl_int bin_max)
     ^~~~~~~~~~~~
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL  -c gpusieve.cpp -o gpusieve.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL  -c perftest.cpp -o perftest.o
perftest.cpp: In function ‘GPUKernels test_cpu_tf_kernels(cl_uint)’:
perftest.cpp:1021:10: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 3 has type ‘cl_ulong’ {aka ‘long unsigned int’} [-Wformat=]
   printf("exponent=%u, %lldM FCs (sieved: %lldM FCs) each, ",
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     mystuff.exponent, num_fcs >> 20, ((cl_ulong)num_loops*mystuff.threads_per_grid)>>20);
                       ~~~~~~~~~~~~~
perftest.cpp:1021:10: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 4 has type ‘cl_ulong’ {aka ‘long unsigned int’} [-Wformat=]
perftest.cpp:1025:10: warning: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 2 has type ‘cl_ulong’ {aka ‘long unsigned int’} [-Wformat=]
   printf("k=%llu, %f GHz-days (assignment), %f GHz-days (per test): ", k, ghzd, ghzdt); fflush(stdout);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~
perftest.cpp: In function ‘GPUKernels test_gpu_tf_kernels(cl_uint)’:
perftest.cpp:1157:10: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 3 has type ‘cl_ulong’ {aka ‘long unsigned int’} [-Wformat=]
   printf("exponent=%u, %lldM FCs each, ", mystuff.exponent, num_fcs>>20);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                    ~~~~~~~~~~~
perftest.cpp:1160:10: warning: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 2 has type ‘cl_ulong’ {aka ‘long unsigned int’} [-Wformat=]
   printf("k=%llu, %f GHz-days (assignment), %f GHz-days (per test): ", k, ghzd, ghzdt); fflush(stdout);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~
perftest.cpp: In function ‘void CL_test(cl_int)’:
perftest.cpp:1696:82: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
   commandQueue = clCreateCommandQueue(context, devices[devnumber], props, &status);
                                                                                  ^
In file included from perftest.cpp:28:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
perftest.cpp:1696:82: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
   commandQueue = clCreateCommandQueue(context, devices[devnumber], props, &status);
                                                                                  ^
In file included from perftest.cpp:28:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
perftest.cpp:1700:84: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
     commandQueue = clCreateCommandQueue(context, devices[devnumber], props, &status);
                                                                                    ^
In file included from perftest.cpp:28:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
perftest.cpp:1700:84: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
     commandQueue = clCreateCommandQueue(context, devices[devnumber], props, &status);
                                                                                    ^
In file included from perftest.cpp:28:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
perftest.cpp:1711:85: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
   commandQueuePrf = clCreateCommandQueue(context, devices[devnumber], props, &status);
                                                                                     ^
In file included from perftest.cpp:28:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
perftest.cpp:1711:85: warning: ‘_cl_command_queue* clCreateCommandQueue(cl_context, cl_device_id, cl_command_queue_properties, cl_int*)’ is deprecated [-Wdeprecated-declarations]
   commandQueuePrf = clCreateCommandQueue(context, devices[devnumber], props, &status);
                                                                                     ^
In file included from perftest.cpp:28:
/opt/rocm/opencl/include/CL/cl.h:1364:1: note: declared here
 clCreateCommandQueue(cl_context                     /* context */,
 ^~~~~~~~~~~~~~~~~~~~
perftest.cpp:1771:3: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
   if (mystuff.CompileOptions[0])  // if mfakto.ini defined compile options, override the default with them
   ^~
perftest.cpp:1774:5: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
     printf("Compiling kernels (build options: \"%s\").", program_options);
     ^~~~~~
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL  -c menu.cpp -o menu.o
gcc -m64 -Wall -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -I/opt/rocm/opencl/include -DBUILD_OPENCL  -c kbhit.cpp -o kbhit.o
g++ sieve.o timer.o parse.o read_config.o mfaktc.o checkpoint.o signal_handler.o filelocking.o output.o mfakto.o gpusieve.o perftest.o menu.o kbhit.o -m64  -O3 -funroll-loops  -ffast-math -finline-functions -frerun-loop-opt -fgcse-sm -fgcse-las -flto -L/opt/rocm/opencl/lib/x86_64 -lOpenCL -o ../mfakto
mfakto.cpp: In function ‘tf_class_opencl.constprop’:
mfakto.cpp:2701:35: note: The ABI for passing parameters with 32-byte alignment has changed in GCC 4.6
           status = run_gs_kernel15(kernel_info[use_kernel].kernel, numblocks, shared_mem_required, k_base, b_in, shiftcount);
                                   ^
mfakto.cpp: In function ‘run_gs_kernel15’:
mfakto.cpp:2112:5: note: The ABI for passing parameters with 32-byte alignment has changed in GCC 4.6
 int run_gs_kernel15(cl_kernel kernel, cl_uint numblocks, cl_uint shared_mem_required, int75 k_base, cl_uint8 b_in, cl_uint shiftcount)
     ^
read_config.c: In function ‘my_read_string’:
read_config.c:124:9: warning: ‘strncpy’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
         strncpy(string, buf + idx + 1, found);
         ^
read_config.c:120:30: note: length computed here
       found = (unsigned int) strlen(buf + idx + 1);
                              ^
cp barrett15.cl ..
cp barrett.cl ..
cp common.cl ..
cp gpusieve.cl ..
cp mfakto_Kernels.cl ..
cp montgomery.cl ..
cp mul24.cl ..
cp datatypes.h ..
cp tf_debug.h ..
cp mfakto.ini ..

```

**Comment by valeriob01** [comment]
After the last commits I have been unable to run mfakto, even I have reinstalled my machine with Debian 10.1 (kernel 4.19) and ROCm 3.3.

**Comment by valeriob01** [comment]
https://github.com/RadeonOpenCompute/ROCm/issues/1103

**Comment by valeriob01** [comment]
Resolved by downgrading to ROCm 3.1
